File: languages/go/client/entity/decrypter.go
package entity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type Decrypter interface {
	ToProto() *proto.Decrypter
}

type AesDecrypter struct {
	Alg  proto.EncryptionAlg
	Args DecrypterArgs
}

func NewAesDecrypter(password string) AesDecrypter {
	return AesDecrypter{
		Alg: proto.EncryptionAlg_A256GCM,
		Args: DecrypterArgs{
			Password: password,
		},
	}
}

func (e AesDecrypter) ToProto() *proto.Decrypter {
	return &proto.Decrypter{
		Alg:  e.Alg,
		Args: e.Args.ToProto(),
	}
}

type DecrypterArgs struct {
	Password string
}

func (e DecrypterArgs) ToProto() *proto.DecrypterArgs {
	return &proto.DecrypterArgs{
		Password: e.Password,
	}
}

File: languages/java/lib/src/main/java/bloock/sdk/java/entity/Decrypter.java
package bloock.sdk.java.entity;

import bloock.sdk.java.bridge.proto.RecordOuterClass;
import bloock.sdk.java.bridge.proto.RecordOuterClass.EncryptionAlg;

public interface Decrypter {
  RecordOuterClass.Decrypter toProto();
}

class AesDecrypter implements Decrypter {
  EncryptionAlg alg;
  DecrypterArgs args;

  AesDecrypter(String password) {
    this.alg = EncryptionAlg.A256GCM;
    this.args = new DecrypterArgs(password);
  }

  @Override
  public RecordOuterClass.Decrypter toProto() {
    return RecordOuterClass.Decrypter.newBuilder()
        .setAlg(this.alg)
        .setArgs(this.args.toProto())
        .build();
  }
}

class DecrypterArgs {
  String password;

  DecrypterArgs(String password) {
    this.password = password;
  }

  RecordOuterClass.DecrypterArgs toProto() {
    return RecordOuterClass.DecrypterArgs.newBuilder().setPassword(this.password).build();
  }
}
